src/components/WalletModal/index.tsx

```diff
@@ -126,101 +119,109 @@ export default function WalletModal({
+  }, [setWalletView])
 
-  const resetAccountView = useCallback(() => {
-    setPendingError(false)
-    setWalletView(WALLET_VIEWS.ACCOUNT)
-  }, [setPendingError, setWalletView])
-
-  // close on connection, when logged out before
   useEffect(() => {
-    if (account && !previousAccount && walletModalOpen) {
-      toggleWalletModal()
+    if (walletModalOpen) {
+      setWalletView(connector === network ? WALLET_VIEWS.OPTIONS : WALLET_VIEWS.ACCOUNT)
     }
-  }, [account, previousAccount, toggleWalletModal, walletModalOpen])
+  }, [walletModalOpen, setWalletView, connector])
 
-  // always reset to account view
   useEffect(() => {
-    if (walletModalOpen) {
-      resetAccountView()
+    if (pendingConnector && walletView !== WALLET_VIEWS.PENDING) {
+      updateWalletError({ wallet: getWalletForConnector(pendingConnector), error: undefined })
+      setPendingConnector(undefined)
     }
-  }, [walletModalOpen, resetAccountView])
+  }, [pendingConnector, walletView])
+
+  const tryActivation = useCallback(
+    async (connector: Connector) => {
+      const name = Object.values(SUPPORTED_WALLETS).find(
+        (supportedWallet) => connector === supportedWallet.connector
+      )?.name
+
+      // log selected wallet
+      sendEvent({
+        category: 'Wallet',
+        action: 'Change Wallet',
+        label: name,
+      })
 
-  const tryActivation = async (connector: AbstractConnector | undefined) => {
-    let name = ''
-    Object.keys(SUPPORTED_WALLETS).map((key) => {
-      if (connector === SUPPORTED_WALLETS[key].connector) {
-        return (name = SUPPORTED_WALLETS[key].name)
-      }
-      return true
-    })
-    // log selected wallet
-    sendEvent({
-      category: 'Wallet',
-      action: 'Change Wallet',
-      label: name,
-    })
-    setPendingWallet(connector) // set wallet for pending view
-    setWalletView(WALLET_VIEWS.PENDING)
+      const wallet = getWalletForConnector(connector)
 
-    // if the connector is walletconnect and the user has already tried to connect, manually reset the connector
-    if (connector instanceof WalletConnectConnector) {
-      connector.walletConnectProvider = undefined
-    }
+      try {
+        // Fortmatic opens it's own modal on activation to log in. This modal has a tabIndex
+        // collision into the WalletModal, so we special case by closing the modal.
+        if (connector === fortmatic) {
+          toggleWalletModal()
+        }
+
+        setPendingConnector(connector)
+        setWalletView(WALLET_VIEWS.PENDING)
+
+        await connector.activate()
 
-    connector &&
-      activate(connector, undefined, true).catch((error) => {
-        if (error instanceof UnsupportedChainIdError) {
-          activate(connector) // a little janky...can't use setError because the connector isn't set
-        } else {
-          setPendingError(true)
+        dispatch(updateSelectedWallet({ wallet }))
+        dispatch(updateWalletError({ wallet, error: undefined }))
```

Comment:
The wallet error should be reset before attempting to activate, or old errors will persist in the UI during attempted activation. Move L180 to L176.